src/intern/drw_textcodec.cpp

```diff
@@ -17,37 +18,67 @@ DRW_TextCodec::DRW_TextCodec()
 
 DRW_TextCodec::~DRW_TextCodec() = default;
 
-void DRW_TextCodec::setVersion(int v, bool dxfFormat){
-    if (v == DRW::AC1009 || v == DRW::AC1006) {
-        version = DRW::AC1009;
-        cp = "ANSI_1252";
-        setCodePage(&cp, dxfFormat);
-    } else if (v == DRW::AC1012 || v == DRW::AC1014
-             || v == DRW::AC1015 || v == DRW::AC1018) {
-        version = DRW::AC1015;
-//        if (cp.empty()) { //codepage not set, initialize
+void DRW_TextCodec::setVersion(DRW::Version v, bool dxfFormat){
+    switch (v)
+    {
+        case DRW::UNKNOWNV:
+        case DRW::MC00:
+        case DRW::AC12:
+        case DRW::AC14:
+        case DRW::AC150:
+        case DRW::AC210:
+        case DRW::AC1002:
+        case DRW::AC1003:
+        case DRW::AC1004:
+            // unhandled?
+            break;
+
+        case DRW::AC1006:
+        case DRW::AC1009:
+        {
+            version = DRW::AC1009;
             cp = "ANSI_1252";
             setCodePage(&cp, dxfFormat);
-//        }
-    } else {
-        version = DRW::AC1021;
-        if (dxfFormat)
-            cp = "UTF-8";//RLZ: can be UCS2 or UTF-16 16bits per char
-        else
-            cp = "UTF-16";//RLZ: can be UCS2 or UTF-16 16bits per char
-        setCodePage(&cp, dxfFormat);
-    }
+            break;
+        }
+
+        case DRW::AC1012:
+        case DRW::AC1014:
+        case DRW::AC1015:
+        case DRW::AC1018:
+        {
+            version = DRW::AC1015;
+    //        if (cp.empty()) { //codepage not set, initialize
+                cp = "ANSI_1252";
+                setCodePage(&cp, dxfFormat);
+            break;
+        }
+
+        case DRW::AC1021:
+        case DRW::AC1024:
+        case DRW::AC1027:
+        case DRW::AC1032:
+        {
+            version = DRW::AC1021;
+            if (dxfFormat)
+                cp = "UTF-8";//RLZ: can be UCS2 or UTF-16 16bits per char
+            else
+                cp = "UTF-16";//RLZ: can be UCS2 or UTF-16 16bits per char
+            setCodePage(&cp, dxfFormat);
+            break;
+        }
+        }
 }
 
-void DRW_TextCodec::setVersion(std::string *v, bool dxfFormat){
-    std::string versionStr = *v;
-    if (versionStr == "AC1009" || versionStr == "AC1006") {
-        setVersion(DRW::AC1009, dxfFormat);
-    } else if (versionStr == "AC1012" || versionStr == "AC1014"
-             || versionStr == "AC1015" || versionStr == "AC1018") {
-        setVersion(DRW::AC1015, dxfFormat);
-    } else {
-        setVersion(DRW::AC1021, dxfFormat);
+void DRW_TextCodec::setVersion(const std::string &v, bool dxfFormat){
+    version = DRW::UNKNOWNV;
+    for ( auto it = DRW::dwgVersionStrings.begin(); it != DRW::dwgVersionStrings.end(); ++it )
```

Comment:
the new logic isn't equivalent to the previous one. For example "AC1006" was mapped to DRW::AC10009, whereas now it will be on DRW::AC1006. Is that intended ?

Reply:
The version from the string is immediately passed to `DRW_TextCodec::setVersion(DRW::Version v, bool dxfFormat)`, which also implements the 1006/1009 -> 1009 mapping. This logic was in two places before, now it's all in `DRW_TextCodec::setVersion(DRW::Version v, bool dxfFormat)`